### ISSUE TITLE
Move TCP socket to node-xmpp-client

### ISSUE THREAD

**Issue by sonnyp** [node-xmpp/node-xmpp-core#58]
I'd like to move TCP socket to node-xmpp-client.

What do you think?

**Comment by chris-rock** [comment]
Any specific reason?

**Comment by sonnyp** [comment]
@chris-rock it makes more sense to me.

We have a BOSH and WebSocket class we should create a TCP class there.
Also, I'd like to reduce the number of useless code for the other pieces of node-xmpp (server, ... )

**Comment by chris-rock** [comment]
makes sense ;-) was just curious. I think it is good to expect a socket, but we do not care about the implementation.

**Comment by sonnyp** [comment]
Only issue is that we'll need to setup a socket on node-xmpp-component but it's basically just gonna be ```this.socket = net.createSocket()```

**Comment by lloydwatkin** [comment]
Sounds good.
